ddl/ddl_api.go

```diff
@@ -2690,6 +2690,10 @@ func (d *ddl) getModifiableColumnJob(ctx sessionctx.Context, ident ast.Ident, or
 	if !mysql.HasAutoIncrementFlag(col.Flag) && mysql.HasAutoIncrementFlag(newCol.Flag) {
 		return nil, errUnsupportedModifyColumn.GenWithStackByArgs("set auto_increment")
 	}
+	// Disallow modifying column from auto_increment to not auto_increment if the session variable `AllowDropAutoInc` is false.
+	if !ctx.GetSessionVars().AllowDropAutoInc && mysql.HasAutoIncrementFlag(col.Flag) && !mysql.HasAutoIncrementFlag(newCol.Flag) {
+		return nil, errUnsupportedModifyColumn.GenWithStackByArgs("drop auto_increment column attribute")
```

Comment:
How about adding a tip in the error message that setting `tidb_allow_drop_auto_inc` if the user truly want to remove the attribute

Reply:
Good idea.